app/services/concerns/xccdf/hosts.rb

```diff
@@ -9,20 +9,40 @@ def save_host
       @host.update!(name: inventory_host['fqdn'])
     end
 
-    def save_profile_host
-      test_result_profile.clone_to(account: @account, host: @host)
+    def host_profile
+      @host_profile ||= test_result_profile.clone_to(
+        account: @account, host: @host
+      )
     end
+    alias save_profile_host host_profile
 
     def report_host
-      @metadata&.dig('fqdn') || @test_result_file.host
+      @metadata&.dig('fqdn') || @test_result_file.test_result.host
+    end
+
+    def associate_rules_from_rule_results
+      ::ProfileRule.import!(
+        ::Rule.where(
+          ref_id: selected_op_rule_results.map(&:id),
+          benchmark_id: @benchmark.id
+        ).pluck(:id).map do |rule_id|
+          ::ProfileRule.new(profile_id: @host_profile.id,
+                            rule_id: rule_id)
+        end, ignore: true
+      )
     end
 
     private
 
     def test_result_profile
       ::Profile.canonical
-               .find_by(ref_id: @test_result_file.test_result.profile_id,
-                        benchmark: @benchmark)
+               .where(ref_id: @test_result_file.test_result.profile_id,
+                      benchmark: @benchmark).first ||
```

Comment:
@dLobatog This `||` condition checks that there's no existing canonical profile.